groundwork/patterns/gw_recipes_pattern.py

```diff
@@ -227,32 +237,53 @@ def build(self, output_dir=None, no_input=False, extra_context=None, **kwargs):
         :param extra_context: A dictionary of context that overrides default and user configuration.
         :return: location of the installed recipe
         """
+
         if output_dir is None:
             output_dir = os.getcwd()
 
         if type(no_input) is not bool:
-            raise IncorrectParameterTypeException('datatype for no_input is not correct')
+            raise IncorrectParameterTypeException('Datatype for no_input is not correct')
 
         if no_input is True:
             if type(extra_context) is not dict and extra_context is not None:
-                raise IncorrectParameterTypeException('datatype for extra_context is not correct')
+                raise IncorrectParameterTypeException('Datatype for extra_context is not correct')
 
         if no_input is False:
             extra_context = None
 
         no_input = no_input
         extra_context = extra_context
 
-        target = cookiecutter(self.path,
-                              output_dir=output_dir,
-                              no_input=no_input,
-                              extra_context=extra_context,
-                              **kwargs)
+        if self.pre_build_validation is not None:
+            if self.pre_build_validation():
```

Comment:
No need to put the cookivutter-build logic inside the if-clause.
Just check like:
```suggestion
          if self.pre_hook is not None and not self.pre_hook(): 
             raise Exception(...)
```
and go on normally. Same for post_hook